arduino-ide-extension/src/browser/serial/monitor/serial-monitor-send-input.tsx

```diff
@@ -110,8 +158,18 @@ export class SerialMonitorSendInput extends React.Component<
     if (keyCode) {
       const { key } = keyCode;
       if (key === Key.ENTER) {
+        const { text } = this.state;
         this.onSend();
+        if (text) {
+          this.state.history.push(text);
+        }
+      } else if (key === Key.ARROW_UP) {
+        this.setState({ text: this.state.history.previous() });
+      } else if (key === Key.ARROW_DOWN) {
+        this.setState({ text: this.state.history.next() });
+      } else if (key === Key.ESCAPE) {
+        this.setState({ text: '' });
       }
     }
   }
-}
+}
```

Comment:
_No newline at end of file_

Please fix it.

Reply:
Newline added.